src/test/librbd/test_librbd.cc

```diff
@@ -2172,7 +2172,33 @@ static void remove_full_try(rados_ioctx_t ioctx, const std::string& image_name,
 
   ASSERT_EQ(0, rbd_close(image));
 
-  // make sure we have latest map that marked the pool full
+  // If using a separate data pool, also make the metadata pool full
+  char pool_name[80];
+  ASSERT_EQ(strlen(pool_name),
```

Comment:
This relies on the order of evaluation and therefore fragile -- we could even crash if the compiler chooses to do `strlen(pool_name)` first, on `pool_name` which isn't initialized.  I'd suggest to relax the assert, using something like `ASSERT_GT(..., 0)`.